---
title: "RNA-seq data preparation"
author: "Jinliang Yang, Ross-Ibarra lab"
date: "February 17, 2015"
output: html_document
---

### Download RNA-seq data

All the RNA-seq data used in this class were downloaded from `NCBI` Sequence Read Archive (SRA) database using [wget](http://www.gnu.org/software/wget/wget.html).
Note: the RNA-seq data and intermediate outputs are put in the folder of `largedata`.


```{r, eval=FALSE}
cd glumipatula/
### Oryza glumipatula, leaf RNA-seq, PE, 101bp
wget get ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR117/SRR1174772/SRR1174772.sra
### Oryza glumipatula, panicle RNA-seq, PE, 101bp
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR117/SRR1174773/SRR1174773.sra
### Oryza glumipatula, root RNA-seq, PE, 101bp
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR117/SRR1174777/SRR1174777.sra

### dump the SRA into fastq
fastq-dump --split-spot --split-3 -A glumipatula/SRR1174772.sra
fastq-dump --split-spot --split-3 -A glumipatula/SRR1174777.sra
fastq-dump --split-spot --split-3 -A glumipatula/SRR1174773.sra
```


### Simulate an experiment with three replications

The aim of this experiment is to compare the differentially expressed genes in two different tissue type: `leaf` vs. `root`. One million reads were extracted for each rep.

```{r, eval=FALSE}
cd largedata/
### checking the total number of reads
wc -l glumipatula/SRR1174772.sra_1.fastq 
#660468896 SRR1170742.sra_1.fastq

### split the fq files into three files as arbitrarily assigned replications
cat glumipatula/SRR1174772.sra_1.fastq | awk 'NR >= 40000001  && NR <= 120000000 { print }' > leaf/leaf.rep1_1.fastq
cat glumipatula/SRR1174772.sra_2.fastq | awk 'NR >= 40000001  && NR <= 120000000 { print }' > leaf/leaf.rep1_2.fastq
cat glumipatula/SRR1174772.sra_1.fastq | awk 'NR >= 140000001  && NR <= 220000000 { print }' > leaf/leaf.rep2_1.fastq
cat glumipatula/SRR1174772.sra_2.fastq | awk 'NR >= 140000001  && NR <= 220000000 { print }' > leaf/leaf.rep2_2.fastq
cat glumipatula/SRR1174772.sra_1.fastq | awk 'NR >= 240000001  && NR <= 320000000 { print }' > leaf/leaf.rep3_1.fastq
cat glumipatula/SRR1174772.sra_2.fastq | awk 'NR >= 240000001  && NR <= 320000000 { print }' > leaf/leaf.rep3_2.fastq

wc -l glumipatula/SRR1174777.sra_1.fastq 
#665118764 SRR1174777.sra_1.fastq
### split the fq files into three files as arbitrarily assigned replications
cat glumipatula/SRR1174777.sra_1.fastq | awk 'NR >= 40000001  && NR <= 120000000 { print }' > root/root.rep1_1.fastq
cat glumipatula/SRR1174777.sra_2.fastq | awk 'NR >= 40000001  && NR <= 120000000 { print }' > root/root.rep1_2.fastq
cat glumipatula/SRR1174777.sra_1.fastq | awk 'NR >= 140000001  && NR <= 220000000 { print }' > root/root.rep2_1.fastq
cat glumipatula/SRR1174777.sra_2.fastq | awk 'NR >= 140000001  && NR <= 220000000 { print }' > root/root.rep2_2.fastq
cat glumipatula/SRR1174777.sra_1.fastq | awk 'NR >= 240000001  && NR <= 320000000 { print }' > root/root.rep3_1.fastq
cat glumipatula/SRR1174777.sra_2.fastq | awk 'NR >= 240000001  && NR <= 320000000 { print }' > root/root.rep3_2.fastq

### split the fq files into three files as arbitrarily assigned replications
cat glumipatula/SRR1174773.sra_1.fastq | awk 'NR >= 40000001  && NR <= 120000000 { print }' > panicle/panicle.rep1_1.fastq
cat glumipatula/SRR1174773.sra_2.fastq | awk 'NR >= 40000001  && NR <= 120000000 { print }' > panicle/panicle.rep1_2.fastq
cat glumipatula/SRR1174773.sra_1.fastq | awk 'NR >= 140000001  && NR <= 220000000 { print }' > panicle/panicle.rep2_1.fastq
cat glumipatula/SRR1174773.sra_2.fastq | awk 'NR >= 140000001  && NR <= 220000000 { print }' > panicle/panicle.rep2_2.fastq
cat glumipatula/SRR1174773.sra_1.fastq | awk 'NR >= 240000001  && NR <= 320000000 { print }' > panicle/panicle.rep3_1.fastq
cat glumipatula/SRR1174773.sra_2.fastq | awk 'NR >= 240000001  && NR <= 320000000 { print }' > panicle/panicle.rep3_2.fastq


```

### Download data for Sativa
```{r, eval=FALSE}
### leaf
cd largedata/sativa/leaf/
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR447/SRR447122/SRR447122.sra
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR447/SRR447123/SRR447123.sra
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR447/SRR447124/SRR447124.sra
fastq-dump --split-spot --split-3 SRR447124.sra
### root
cd ../root/
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR447/SRR447117/SRR447117.sra
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR447/SRR447118/SRR447118.sra
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR447/SRR447119/SRR447119.sra
fastq-dump --split-spot --split-3 SRR447117.sra
fastq-dump --split-spot --split-3 SRR447118.sra
fastq-dump --split-spot --split-3 SRR447119.sra
### panicle PE 75bp
cd ../panicle/
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR037/SRR037739/SRR037739.sra
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR037/SRR037738/SRR037738.sra
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/SRR/SRR037/SRR037734/SRR037734.sra
fastq-dump --split-spot --split-3 SRR037734.sra
fastq-dump --split-spot --split-3 SRR037738.sra
fastq-dump --split-spot --split-3 SRR037739.sra
```




### Log sample info to a txt file

Open `vi`, specify the **relative location** and names of the fastq files.
And, the `replications` and `treatments` of your experimental design.

```{r, eval=FALSE}
### copy the following text into your sample.txt file.
fq1 fq2 rep tissue taxa
glumipatula/leaf/leaf.rep1_1.fastq glumipatula/leaf/leaf.rep1_2.fastq rep1 leaf glumipatula
glumipatula/leaf/leaf.rep2_1.fastq glumipatula/leaf/leaf.rep2_2.fastq rep2 leaf glumipatula
glumipatula/leaf/leaf.rep3_1.fastq glumipatula/leaf/leaf.rep3_2.fastq rep3 leaf glumipatula

fq1 fq2 rep tissue taxa
glumipatula/root/root.rep1_1.fastq glumipatula/root/root.rep1_2.fastq rep1 root glumipatula
glumipatula/root/root.rep2_1.fastq glumipatula/root/root.rep2_2.fastq rep2 root glumipatula
glumipatula/root/root.rep3_1.fastq glumipatula/root/root.rep3_2.fastq rep3 root glumipatula

```


```{r, eval=FALSE}
### sample2
fq1 fq2 rep tissue taxa
glumipatula/panicle/panicle.rep1_1.fastq glumipatula/panicle/panicle.rep1_2.fastq rep1 panicle glumipatula
glumipatula/panicle/panicle.rep2_1.fastq glumipatula/panicle/panicle.rep2_2.fastq rep2 panicle glumipatula
glumipatula/panicle/panicle.rep3_1.fastq glumipatula/panicle/panicle.rep3_2.fastq rep3 panicle glumipatula
```

./leaf:
SRR447122_1.fastq  SRR447122.sra      SRR447123_2.fastq  SRR447123.fastq  SRR447124_1.fastq  SRR447124.sra
SRR447122_2.fastq  SRR447123_1.fastq  SRR447123_.fastq   SRR447123.sra    SRR447124_2.fastq

./panicle:
SRR037734_1.fastq  SRR037734.sra      SRR037738_2.fastq  SRR037739_1.fastq  SRR037739.sra
SRR037734_2.fastq  SRR037738_1.fastq  SRR037738.sra      SRR037739_2.fastq

```{r, eval=FALSE}
### samples for sativa
fq1 fq2 rep tissue taxa
sativa/leaf/SRR447122_1.fastq sativa/leaf/SRR447122_2.fastq rep1 leaf sativa
sativa/leaf/SRR447123_1.fastq sativa/leaf/SRR447123_2.fastq rep2 leaf sativa
sativa/leaf/SRR447124_1.fastq sativa/leaf/SRR447124_2.fastq rep3 leaf sativa

fq1 fq2 rep tissue taxa
sativa/root/SRR447117_1.fastq sativa/root/SRR447117_2.fastq rep1 root sativa
sativa/root/SRR447118_1.fastq sativa/root/SRR447118_2.fastq rep2 root sativa
sativa/root/SRR447119_1.fastq sativa/root/SRR447119_2.fastq rep3 root sativa

fq1 fq2 rep tissue taxa
sativa/panicle/SRR037734_1.fastq sativa/panicle/SRR037734_2.fastq rep1 leaf sativa
sativa/panicle/SRR037738_1.fastq sativa/panicle/SRR037738_2.fastq rep2 leaf sativa
sativa/panicle/SRR037739_1.fastq sativa/panicle/SRR037739_2.fastq rep3 leaf sativa
```



### Download reference genome and annotation data

Two reference genomes for rice are available. __Oryza sativa__ indica is one of the them. The genome assembly and gene annotation can be downloaded from [EnsemblPlants](http://plants.ensembl.org/Oryza_indica/Info/Index).

```{r, eval=FALSE}
mkdir OS_indica
cd OS_indica
wget ftp://ftp.ensemblgenomes.org/pub/plants/release-25/fasta/oryza_indica/dna/Oryza_indica.ASM465v1.25.dna.chromosome.*.fa.gz
```


### Setup alignment database

[GSNAP (Genomic Short-read Nucleotide Alignment Program)](http://www.ncbi.nlm.nih.gov/pubmed/20147302) is a fast and splice-aware aligner. You can download and install it from [here](http://research-pub.gene.com/gmap/). 

```{r, eval=FALSE}
### load the module of gmap if you use farm
module load gmap/2014-05-15
### http://research-pub.gene.com/gmap/src/README
### Setting up to build a GMAP/GSNAP database (one chromosome per FASTA entry)
### Note: be careful about your dir specified by -D
gmap_build -D largedata/OS_indica/ -d ASM465v1.25_gsnap -g Oryza_indica.ASM465v1.25.dna.chromosome*fa.gz > gmapbuild.log  

```
