---
title: "Short Reads Processing"
author: "Jinliang Yang"
date: "February 17, 2015"
output: html_document
---

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r, eval=FALSE}

source("profiling/1.RNA-seq/1.A.1_run_alignment.R")


```


### prepare genomic features

```{r, eval=FALSE}
source("http://bioconductor.org/biocLite.R")
biocLite("GenomicFeatures")
library(GenomicFeatures)

txdb <- makeTranscriptDbFromGFF(
  file="data/Osativa_204_v7.0.gene_exons",
  format="gff3", species="oryza_indica",useGenesAsTranscripts=FALSE,
  dataSource="http://genome.jgi.doe.gov/pages/dynamicOrganismDownload.jsf?organism=Osativa")

saveDb(txdb, file="cache/Osativa_204_v7.0.sqlite")
```

You can also embed plots, for example:

```{r, echo=FALSE}
plot(cars)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
